### PR TITLE
Roaches have armor

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Moves armor from golems to superior animals, gives roaches armor.
Adds AP support to superior animals, giving some to jager, fuhrer and kaiser, and giving BS roaches quite a large amount.

## Why It's Good For The Game

Roaches are more varied, are resistant to different damage types, and will eat through light armour a lot better. Beware the roach.

## Changelog
:cl:
balance: Roaches buffed, have armour, and penetrate armour. Fear the swarm.
tweak: Golems have a separate minimum and maximum damage
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/superior_animal/defense.dm [hunk 1]
```diff
@@ -379,3 +379,21 @@
...
+	if(istype(user, /mob/living))
+		var/mob/living/L = user
+		L.armor_penetration = penetration
+
+	damage_through_armor(damage, BRUTE, attack_flag=ARMOR_MELEE, armour_pen=penetration)
+	user.attack_log += text("\[[time_stamp()]\] <font color='red'>attacked [src.name] ([src.ckey])</font>")
+	src.attack_log += text("\[[time_stamp()]\] <font color='orange'>was attacked by [user.name] ([user.ckey])</font>")
+	src.visible_message(SPAN_DANGER("[user] has [attack_message] [src]!"))
+	user.do_attack_animation(src)
+	spawn(1) updatehealth()
```
Comment (MLGTASTICa) [inline code/modules/mob/living/carbon/superior_animal/defense.dm:398]: why the spawn 1?

**Review by SirRichardFrancis — COMMENTED** [review]
Why roaches have halloss handling at all? Aren't they taking any damage as real damage?
Adding armor to fat roaches kind of makes sense, but I don't like the code at all.
Radiation and bio resistance? Damage and AP buff?
100% armor penetration and 80% bullet resistance, for fuck sake...
Missing newlines and `return 1` instead of `return TRUE` too.

File: code/modules/mob/living/carbon/superior_animal/roach/types/toxic.dm [hunk 2]
```diff
@@ -20,6 +20,11 @@
 	acceptableTargetDistance = 5
 	kept_distance = 3
 
+	// Armor related variables
+	bullet = 20
+	bio = 75
+	rad = 50
```
Comment (vode-code) [inline code/modules/mob/living/carbon/superior_animal/roach/types/toxic.dm:26]:
```suggestion
	rad = 100
```
Gestrahltes are radioactive and spit radioactive goo, if they were not immune they would have already died.

File: code/modules/mob/living/carbon/superior_animal/defense.dm (hunk 1 above)
Comment (Humonitarian) [reply code/modules/mob/living/carbon/superior_animal/defense.dm:398]: Ask the person who wrote living_defense.dm

**Comment by Humonitarian** [comment]
> Why roaches have halloss handling at all?
It is converted into real damage. They have halloss handling because it is a separate damage type, and I'm not planning to write extra_superior_animal.dm and refactor the Baycode damage handling.

> Adding armor to fat roaches kind of makes sense, but I don't like the code at all.
Ok

> Radiation and bio resistance? Damage and AP buff? 100% armor penetration and 80% bullet resistance, for fuck sake...
I feel they are justified. Unbekannt roaches are absolutely underwhelming and rarely do anything at all. Not at least they can poke through armor

> Missing newlines and `return 1` instead of `return TRUE` too.
The life of a control+c control+v enjoyer, I'll see to fixing them

**Comment by Kurgis** [comment]
What's the benefit of this versus using both `burn_mod` and `brute_mod`?

**Comment by Humonitarian** [comment]
> What's the benefit of this versus using both `burn_mod` and `brute_mod`?

All those do is make either burn of brute do less damage. Not at all the intended change, as it doesn't take armor penetration, GDR and other variables into account.

File: code/modules/mob/living/carbon/superior_animal/superior_animal.dm [hunk 3]
```diff
@@ -99,6 +99,14 @@
 	var/grabbed_by_friend = FALSE //is this superior_animal being wrangled?
 	var/ticks_processed = 0
 
+	// Armor related variables
+	var/melee = 0
```
Comment (drexample) [inline code/modules/mob/living/carbon/superior_animal/superior_animal.dm:103]: This really ought to be a datum like all other armour handling types.

**Review by drexample — CHANGES_REQUESTED** [review]
The variables aren't intuitive and should be one big datum instead of several vars

**Review by drexample — APPROVED** [review]

**Review by mattroks101 — APPROVED** [review]

**Pull request merged**